asm_riscv/src/lib.rs

```diff
@@ -520,97 +520,96 @@ impl From<I> for u32 {
     }
 }
 
-impl From<u32> for I {
+impl TryFrom<u32> for I {
+    type Error = ();
     // Using match makes it easier to extend code in the future.
     #[allow(clippy::match_single_binding)]
-    fn from(with: u32) -> Self {
+    fn try_from(with: u32) -> Result<Self, Self::Error> {
         match with & 0b1111111 {
             // Load From RAM
             0b0000011 => match I::from_i(with) {
-                (d, 0b000, s, im) => LB { d, s, im },
-                (d, 0b001, s, im) => LH { d, s, im },
-                (d, 0b010, s, im) => LW { d, s, im },
-                (d, 0b100, s, im) => LBU { d, s, im },
-                (d, 0b101, s, im) => LHU { d, s, im },
-                (_, funct, _, _mm) => panic!("Unknown funct3: {}", funct),
+                (d, 0b000, s, im) => Ok(LB { d, s, im }),
+                (d, 0b001, s, im) => Ok(LH { d, s, im }),
+                (d, 0b010, s, im) => Ok(LW { d, s, im }),
+                (d, 0b100, s, im) => Ok(LBU { d, s, im }),
+                (d, 0b101, s, im) => Ok(LHU { d, s, im }),
+                (_, _, _, _mm) => Err(()),
             },
             // Misc. Memory Instructions
             0b0001111 => match I::from_i(with) {
-                (_, 0b000, _, im) => FENCE { im },
-                (_, funct, _, _mm) => panic!("Unknown funct3: {}", funct),
+                (_, 0b000, _, im) => Ok(FENCE { im }),
+                (_, _, _, _mm) => Err(()),
             },
             // Store To RAM
             0b0100011 => match I::from_s(with) {
-                (0b000, s1, s2, im) => SB { s1, s2, im },
-                (0b001, s1, s2, im) => SH { s1, s2, im },
-                (0b010, s1, s2, im) => SW { s1, s2, im },
-                (funct, _s, _z, _mm) => panic!("Unknown funct3: {}", funct),
+                (0b000, s1, s2, im) => Ok(SB { s1, s2, im }),
+                (0b001, s1, s2, im) => Ok(SH { s1, s2, im }),
+                (0b010, s1, s2, im) => Ok(SW { s1, s2, im }),
+                (_, _s, _z, _mm) => Err(()),
             },
```

Comment:
I think I would like to reduce the number of `Ok`, and make the `Err` stand out with `return`.

```rust
            0b0100011 => Ok(match I::from_s(with) {
                (0b000, s1, s2, im) => SB { s1, s2, im },
                (0b001, s1, s2, im) => SH { s1, s2, im },
                (0b010, s1, s2, im) => SW { s1, s2, im },
                (_, _s, _z, _mm) => return Err(()),
            }),
```

Actually, I think `Ok` would work on the outer `match` statement as well;

```rust
    fn try_from(with: u32) -> Result<Self, Self::Error> {
        Ok(match with & 0b1111111 {
```

Reply:
Good catch, fixed this